### PR TITLE
Update initial `earliest_exit_epoch` calculation & add tests

### PR DESCRIPTION
**Pull request by jtraglia**
This came up when discussing an edge case: what is the correct `earliest_exit_epoch` value given a state where the greatest validator exit epoch is less than the current epoch. For example, given a state with one validator and its exit epoch is 3, current epoch is 100, should `earliest_exit_epoch` be 4? That doesn't seem right.

I believe it should be at least `compute_activation_exit_epoch(get_current_epoch(pre)) + 1`. And updated to `validator.exit_epoch + 1` if there is a validator with a greater exit epoch.

### CODE REVIEW

**Review by ralexstokes — APPROVED** [review]
given the implementation of the exit churn computation post-electra, this PR will have no discernible change

I can see this the _intent_ of `upgrade_to_electra` so fine with me if we want to be explicit about it

File: specs/electra/fork.md [hunk 1]
```diff
@@ -74,10 +74,12 @@ def upgrade_to_electra(pre: deneb.BeaconState) -> BeaconState:
     epoch = deneb.get_current_epoch(pre)
     latest_execution_payload_header = pre.latest_execution_payload_header
 
-    exit_epochs = [v.exit_epoch for v in pre.validators if v.exit_epoch != FAR_FUTURE_EPOCH]
-    if not exit_epochs:
-        exit_epochs = [get_current_epoch(pre)]
-    earliest_exit_epoch = max(exit_epochs) + 1
+    earliest_exit_epoch = compute_activation_exit_epoch(get_current_epoch(pre))
```
Comment (mkalinin) [inline specs/electra/fork.md:77]: maybe:
```python
earliest_exit_epoch = max(
    [v.exit_epoch for v in pre.validators if v.exit_epoch != FAR_FUTURE_EPOCH] +
    [compute_activation_exit_epoch(get_current_epoch(pre))]
)
```
I don’t think the `+= Epoch(1)` is needed since we have `compute_activation_exit_epoch`

Comment (jtraglia) [reply specs/electra/fork.md:77]: Hmm I think that code is a bit too complex/pythonic. Clients would have trouble following this exactly.

I believe the `+= Epoch(1)` is necessary if we want similar functionality as before. `compute_activation_exit_epoch` only sets the lower bound.

Comment (mkalinin) [reply specs/electra/fork.md:77]: Yes, I agree. It should be `+1` to ensure that the exit mechanism is reset in case when the `earliest_exit_epoch` is far in the future. I am happy to leave the code as is and appreciating the argument on not making the spec too pythonic.
